README.md

```diff
@@ -1,12 +1,267 @@
+      * [ui](#packagesui)
+      * [utils](#packagesutils)
+      * [syndesis-context](#packagessyndesis-context)
+  * [typings](#typings)
+* [Installation](#installation)
+* [Building](#building)
+* [Scripts](#scripts)
+* [Roadmap](#roadmap)
+
+## Architecture 
+
+We use [Lerna](https://github.com/lerna/lerna) to streamline the development process; the most common operations, 
+like building the project or running the development mode can be done directly from the project root.  
+
+Code is split in many packages, organized as a monorepo using [Yarn's workspaces](https://yarnpkg.com/lang/en/docs/workspaces/). 
+The workspace is configured like this:
+
 ```
+"syndesis",
+"packages/*"
+"typings/*"
+```
+
+### syndesis
+
+`syndesis` is the main application. It handles the authentication against Syndesis's OAuth Server, and provides the main
+ app layout where "sub-apps" can be injected. 
+
+![](doc/assets/syndesis-chrome.png)
+
+It also provides an [API for sub-apps](#packagessyndesis-context) - in the form of a [React's Context](https://reactjs.org/docs/context.html) - for 
+interact with it, eg. closing the navigation bar, or redirecting to another sub-app.
+
+It's built with [create-react-app](https://github.com/facebook/create-react-app).
+
+#### Development server
+
+```bash
+# From the repository root
+$ yarn watch:app
+```
+
+Or
+
+```bash
+# From the syndesis folder
+$ yarn start
+```
+
+
+#### First time setup
+
+The app requires some extra configuration on your Syndesis installation in order to work.
+
+We need to create an OAuth Client to allow the UI to login against Syndesis OAuth Server, and then expose the 
+server with a direct route.
+
+First, login as an "admin" user of your cluster:    
+```bash
+$ oc login -u admin
+```
+
+Then, create a new OAuth Client:
+```bash
 oc create -f <(echo '
 kind: OAuthClient
 apiVersion: oauth.openshift.io/v1
 metadata:
  name: camel-k-ui
 secret: "..."
 redirectURIs:
- - "http://localhost:3000"
+ - "http://localhost:5000/"
 grantMethod: prompt
 ')
-```
\ No newline at end of file
+```
+
+Finally, expose the server with a direct route.   
+**Replace the `CLUSTER_ADDRESS` placeholder with the right address of your cluster.**
+For a local development environment running on Minishift, it will look like this: `192.168.64.1.nip.io` 
+
+```bash
+oc create -f <(echo 'apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  annotations:
+    openshift.io/host.generated: "true"
+  labels:
+    app: syndesis
+    syndesis.io/app: syndesis
+    syndesis.io/component: syndesis-server
+    syndesis.io/type: infrastructure
+  name: syndesis-server
+  namespace: syndesis
+spec:
+  host: syndesis-server.CLUSTER_ADDRESS
+  to:
+    kind: Service
+    name: syndesis-server
+    weight: 100
+  wildcardPolicy: None
+')
+```
+
+Now we can setup the app. Run the [syndesis development server](#development-server) and open the address in a browser.  
+You will be asked to provide two information:
```

Comment:
This suggests that I'd be redirected to the settings page upon navigating to the dev server url. I remember seeing this happen before, but now for some reason I don't get automatically redirected to settings and instead the UI partially loads whatever view I'm navigated with the the network requests to connections/integrations failing.

Might be helpful to just mention to navigate to the settings page to enter this information unless we can have the redirects work again?

Reply:
I'm thinking that eventually we'll want to adjust this to use a `config.json` file, which would then be used to specify the API endpoint and oauth endpoints.  The main reason is we already have a bit of infrastructure in place to support this method; the `config.json` file is loaded in the real app via a config map, we could probably use the same approach for these settings.

Reply:
@seanforyou23: the redirect happens only when the settings are empty, so what you experienced is right.

But yeah, the “app” component design is totally in a POC state. As Stan said, we should “talk” with Syndesis using the constructs that are already available, but for the moment I think we can continue with what we have.

Reply:
Ok makes sense, thx for the heads up!